### PR TITLE
Uniformize CssValue.Equals implementation

### PR DESCRIPTION
**Pull request by meziantou**
# Types of Changes

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## Description

Follow up: https://github.com/AngleSharp/AngleSharp.Css/pull/162#discussion_r1502917208

### CODE REVIEW

File: src/AngleSharp.Css/Values/Functions/CssFontFormatValue.cs [hunk 1]
```diff
@@ -39,7 +40,15 @@ public CssFontFormatValue(String fontFormat)
         /// </summary>
         /// <param name="other">The value to check against.</param>
         /// <returns>True if both are equal, otherwise false.</returns>
-        public Boolean Equals(CssFontFormatValue other) => _fontFormat.Equals(other._fontFormat);
+        public Boolean Equals(CssFontFormatValue other)
+        {
+            if (other is not null)
+            {
+                return _fontFormat == other._fontFormat;
```
Comment (FlorianRappl) [inline src/AngleSharp.Css/Values/Functions/CssFontFormatValue.cs]: In a similar scenario we used another construct ([CssAttrValue.cs](https://github.com/AngleSharp/AngleSharp.Css/pull/163/files#diff-30b1230b47b07604c15130f3f2ac5a6a275a934ea26f4fa50ddea61500c9523c)); intentional?

File: src/AngleSharp.Css/Values/Functions/CssRunningValue.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public CssRunningValue(String ident)
         /// </summary>
         /// <param name="other">The value to check against.</param>
         /// <returns>True if both are equal, otherwise false.</returns>
-        public Boolean Equals(CssRunningValue other) => _ident.Equals(other._ident);
+        public Boolean Equals(CssRunningValue other) => other is not null && _ident == other._ident;
```
Comment (FlorianRappl) [inline src/AngleSharp.Css/Values/Functions/CssRunningValue.cs:62]: For this scenario (single primitive string field) we not have 3 different constructs (e.g., [CssAttrValue.cs](https://github.com/AngleSharp/AngleSharp.Css/pull/163/files#diff-30b1230b47b07604c15130f3f2ac5a6a275a934ea26f4fa50ddea61500c9523c)). Intentional?

**Review by FlorianRappl — COMMENTED** [review]
You are a true magician - such a quick one!

I'd say looks great to me - only question is in the commented scenario; it seems all three files are the same yet we use slightly (yet overall, of course, equivalent) constructs.

File: src/AngleSharp.Css/Values/Functions/CssFontFormatValue.cs (hunk 1 above)
Comment (meziantou) [reply src/AngleSharp.Css/Values/Functions/CssFontFormatValue.cs]: I may have missed a few cases, but the change should be
- `ICssValue`: Use the comparer
- `string`: `left.Equals(right)` => `left == right`
- other types: keep it as-is

I'll give it another look

File: src/AngleSharp.Css/Values/Functions/CssRunningValue.cs (hunk 2 above)
Comment (meziantou) [reply src/AngleSharp.Css/Values/Functions/CssRunningValue.cs:62]: I've fixed CssFontFormatValue to be single line

Comment (meziantou) [reply src/AngleSharp.Css/Values/Functions/CssRunningValue.cs:62]: And also `CssAttrValue`

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks!

**Pull request merged**